data-prepper-core/src/test/java/com/amazon/dataprepper/meter/EMFLoggingMeterRegistryTest.java

```diff
@@ -0,0 +1,410 @@
+    }
+
+    @Test
+    void registryMetricsLoggerDropEmptyTag() {
+        registry.gauge("gauge", Tags.of("empty", ""), 1d);
+        final List<MetricsLogger> metricsLoggerList = registry.metricsLoggers();
+        assertThat(metricsLoggerList.size(), equalTo(1));
+        final MetricsLogger metricsLogger = metricsLoggerList.get(0);
+        final MetricsContext context = reflectivelyGetMetricsContext(metricsLogger);
+        assertThat(hasDimension(context, "empty"), is(false));
+    }
+
+    @Test
+    void registryPublish() {
+        final MetricsLogger testMetricsLogger1 = mock(MetricsLogger.class);
+        final MetricsLogger testMetricsLogger2 = mock(MetricsLogger.class);
+        when(registry.metricsLoggers()).thenReturn(Arrays.asList(testMetricsLogger1, testMetricsLogger2));
+        registry.publish();
+        verify(testMetricsLogger1, times(1)).flush();
+        verify(testMetricsLogger2, times(1)).flush();
+    }
+
+    private MetricsContext reflectivelyGetMetricsContext(final MetricsLogger metricsLogger) {
+        final Field field;
+        try {
+            field = metricsLogger.getClass().getDeclaredField("context");
+            field.setAccessible(true);
+            final MetricsContext context = (MetricsContext) field.get(metricsLogger);
+            field.setAccessible(false);
+            return context;
+        } catch (final NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private boolean hasDimension(final MetricsContext context, final String key) {
+        final List<DimensionSet> dimensionSetList = context.getDimensions();
+        for (final DimensionSet dimensionSet:dimensionSetList) {
+            if (dimensionSet.getDimensionKeys().contains(key)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void assertDimensionEqual(final MetricsContext context, final String key, final String value) {
+        final List<DimensionSet> dimensionSetList = context.getDimensions();
+        assertThat(dimensionSetList.size(), equalTo(1));
+        final DimensionSet dimensionSet = dimensionSetList.get(0);
+        assertThat(dimensionSet.getDimensionKeys().size(), equalTo(1));
+        assertThat(dimensionSet.getDimensionValue(key), equalTo(value));
+    }
+
+    private boolean HasMetric(final MetricsContext context, final Id meterId, final String suffix) {
+        try {
+            final List<String> emfJsonList = context.serialize();
+            assertThat(emfJsonList.size(), equalTo(1));
+            final Map<String, Object> emfMap = objectMapper.readValue(emfJsonList.get(0), MAP_TYPE_REFERENCE);
+            String metricName = meterId.getName();
+            if (suffix != null) {
+                metricName = metricName.concat(".").concat(suffix);
+            }
+            return emfMap.get(metricName) instanceof Double;
+        } catch (final JsonProcessingException e) {
```

Comment:
In main code, I do think wrapping checked exceptions is nice. But, in test code, I recommend that you just add the checked exceptions. The code can be a little simpler this way.